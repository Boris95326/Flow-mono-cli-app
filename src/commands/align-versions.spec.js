// @flow

jest.mock('./../lib/paths.js');
jest.mock('./../lib/dependency.js');
jest.mock('./../lib/logger.js');

const path: any = require('./../lib/paths.js');
const dependency: any = require('./../lib/dependency.js');

const alignFlowVersions = require('./align-versions.js');

describe('align-versions', () => {
  it('should export an function', () => {
    expect(typeof alignFlowVersions).toBe('function');
  });

  it('should call the dependency.logVersionMisMatch method for each resolved package path and flow dependencies', async () => {
    path.resolveMonoRepoPackagePaths.mockReturnValue([
      '/foo/bar/package-a',
      '/foo/bar/package-b',
      '/foo/bar/package-c'
    ]);

    await alignFlowVersions();

    expect(dependency.logVersionMisMatch.mock.calls).toMatchSnapshot();
  });
});
