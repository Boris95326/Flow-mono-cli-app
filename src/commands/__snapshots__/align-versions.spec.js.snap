// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`align-versions should call the dependency.logVersionMisMatch method for each resolved package path and flow dependencies 1`] = `
Array [
  Array [
    "flow-bin",
    "/foo/bar/package-a",
  ],
  Array [
    "flow-typed",
    "/foo/bar/package-a",
  ],
  Array [
    "flow-bin",
    "/foo/bar/package-b",
  ],
  Array [
    "flow-typed",
    "/foo/bar/package-b",
  ],
  Array [
    "flow-bin",
    "/foo/bar/package-c",
  ],
  Array [
    "flow-typed",
    "/foo/bar/package-c",
  ],
]
`;
